platform/platform.hpp

```diff
@@ -214,6 +215,9 @@ class Platform
   template <typename Task>
   void RunOnNetworkThread(Task && task) { m_networkThread.Push(forward<Task>(task)); }
 
+  template <typename Task>
```

Comment:
I suggest you to implement following before the Platform:
```c++
enum Thread
{
  File,
  Network,
  Gui
};
```
and then, do following:
```c++
template <typename Task>
void RunTask(Thread thread, Task && task)
{
  switch (thread)
  {
    ...
  }
}
```
This allows you to implement more tricky things, like:
```c++
template <typename Task1, typename Task2>
void RunTaskAndReply(Thread thread1, Task && task1, Thread thread2, Task && task2)
{
   auto task = []()
   {
      auto rv = task1();
      GetPlatform().RunTask(thread2, bind(task2, rv));
   };
   RunTask(thread1, task);
}
```